cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -433,6 +435,7 @@ private void createNamespaces() {
     }
     LOG.info("Creating and registering {} namespace", Constants.DEFAULT_NAMESPACE);
     getStore().createNamespace(Constants.DEFAULT_NAMESPACE_META);
+    locationFactory.create(Constants.DEFAULT_NAMESPACE);
```

Comment:
This will not create the directory. Should call ``mkdirs`` on the object returned by ``create``